### ISSUE TITLE
RSS sources

### ISSUE THREAD

**Issue by Eitot** [ViennaRSS/vienna-rss#1411]
Vienna has these RSSSources.plist files that are used for populating a pop-up menu in the “new subscription” panel:
<img width="425" alt="Screenshot 2020-12-26 at 19 38 38" src="https://user-images.githubusercontent.com/15073177/103157322-f7605080-47b1-11eb-8b89-d1d681841058.png">

Some observations:
- "Local File" and “URL" could be merged into one field, as NSURL is sophisticated enough to distinguish paths from URLs.
- MSN Spaces and Splinder (Italian localisation only) seem to be discontinued/unreachable
- Xanga seems to be reachable (xanga.com), but the currently-used URL does not work anymore
- Blogspot supports `<link>` auto-discovery like any other HTML page
- LiveJournal is still live, but [this page](https://www.livejournal.com/support/faq/149.html) suggests that the API has changed

Suggestion: remove RSSSources.plist and support only local file paths (e.g. `file://path/to/feed.xml`), feed URLs (e.g. `https://hostname/feed.xml`) or auto-discoverable feeds in the `<head>` of HTML pages.

<img width="502" alt="Screenshot 2020-12-26 at 19 32 49" src="https://user-images.githubusercontent.com/15073177/103157232-25916080-47b1-11eb-9883-737512c73758.png">

**Comment by barijaona** [comment]
Improving it and adding support for YouTube would be handy. Cf. [my comment](https://github.com/ViennaRSS/vienna-rss/issues/1347#issuecomment-720115192) in issue #1347

**Comment by Eitot** [comment]
> Improving it and adding support for YouTube would be handy. Cf. [my comment](https://github.com/ViennaRSS/vienna-rss/issues/1347#issuecomment-720115192) in issue #1347

It does seem that YouTube channel pages have a `<link ref="alternate" type="application/rss+xml" href="..." />` element, however, they put it in the `<body>` element, which contradicts the HTML standard. Vienna's auto-discovery feature does not parse anything beyond the closing `</head>` element.

The HTML parser could perhaps be extended cover these edge cases.

**Comment by TAKeanice** [comment]
Have a look at the expression I use in the Browser+RSSSource implementation, that's probably easier than extending HTML parsing

**Comment by barijaona** [comment]
Indeed, the webkit implementation does not limit itself to the `<head>` element and finds the link. We can classify this as a feature :wink:

**Comment by barijaona** [comment]
There are a few more sources in `Resources/zh-Hant.lproj/RSSSources.plist`
Most of these Chinese sources also include `<link>` auto discovery.

But an interesting exception is http://mypaper.pchome.com.tw/
I would not have guessed that the RSS feed for an user is `http://mypaper.pchome.com.tw/USERNAME/rss`

Another interesting example is Tumblr (not implemented in Vienna yet). There is an RSS feed, but it is hidden at `https://USERNAME.tumblr.com/rss`
I see some interesting things on some Tumblrs, but I don't see myself registering into Tumblr to follow these blogs…

If only there was a similar trick for Medium… 😜 

Should we implement user supported plugins here ?

**Comment by Eitot** [comment]
> Have a look at the expression I use in the Browser+RSSSource implementation, that's probably easier than extending HTML parsing

This only works in the context of the browser view though. Vienna uses feed auto-discovery when adding a new feed also. WebKit is not available in that context.

I had been working on a small feed discoverer based on libxml2's HTML parser last year, because the current implementation is all over the place (coupling) and not that robust. I managed to make it work with Swift yesterday, but I need to clean it up some more. I will submit a PR soon. I have made it so that it parses only `<link>` elements, regardless where they are to cover these non-standard cases as YouTube.

> Another interesting example is Tumblr (not implemented in Vienna yet). There is an RSS feed, but it is hidden at `https://USERNAME.tumblr.com/rss`

Tumblr properly uses `<link rel="alternate" type="application/rss+xml" … >` at `https://username.tumblr.com`, so this works with auto-discovery.

> Should we implement user supported plugins here ?

A user-made extension would be my preference. Vienna should then also make some pre-made plug-ins available for users to download, maybe via the webpage.

**Comment by barijaona** [comment]
Many template templates do not expose the link for auto discover.
I also support the user-made extension / plugin solution.

**Comment by DavidMunch** [comment]
Just to add here; When I try to add a new feed, the source drop down doesn't default to any selection, it is just blank. Since Vienna automatically adds the content of the copy/paste function (Does that have a term? It was called the drawer under Mac OS 9 in my language), you can immediately press the Subscribe button, which results in absolutely nothing happening, which is really bad HIG behavior!
I think it should either default to "URL" or the last used option, since people likely have preferences based on their use cases.

**Comment by upstageleft** [comment]
The "local file" option should be replaced with or supplemented by a "local script" source which, when added as a subscription, runs the specified shell script when refreshed. If the output of the script is valid RSS XML, then that should be parsed as normal. This allows the user much more flexibility in customizing their feed content (such as by removing from a feed articles on a specific topic or containing a specific keyword). This also may allow users to correct output from a misconfigured feed generator about which the site operator is unresponsive. This is a feature I have seen in other RSS readers, and which I used consistently but have lost the use of since switching to Vienna.

**Comment by matthewruzzi** [comment]
It also could be helpful to have a built in feature to be able to enter a URL and then scrape data to generate a feed based on XPath or CSS selectors without needing to manually write a shell script.
